src/main/java/org/gridsuite/modification/server/modifications/ShuntCompensatorCreation.java

```diff
@@ -111,7 +111,7 @@ private void createShuntInBusBreaker(VoltageLevel voltageLevel, ShuntCompensator
         voltageLevel.newShuntCompensator()
             .setId(shuntCompensatorInfos.getEquipmentId())
             .setName(shuntCompensatorInfos.getEquipmentName())
-            .setSectionCount(shuntCompensatorInfos.getCurrentNumberOfSections())
+            .setSectionCount(shuntCompensatorInfos.getCurrentNumberOfSections() != null ? shuntCompensatorInfos.getMaximumNumberOfSections() : 1)
```

Comment:
.setSectionCount(shuntCompensatorInfos.getCurrentNumberOfSections() != null ? shuntCompensatorInfos.getCurrentNumberOfSections() : 1)


You can also use : ObjectUtils.defaultIfNull(shuntCompensatorInfos.getCurrentNumberOfSections(), 1);

Reply:
done